### PR TITLE
Major CI improvements

### PR DESCRIPTION
**Pull request by Cadair**
This does the following:

* Enables OS/X testing on Travis
* Fixes Windows testing on Appveyor
* Use conda to install Python deps on both Travis and Appveyor for all OSes
* Marks the online build as an 'allowed faliure' so that it will report a green status even if there are server issues. This build should be manually checked for online heavy PRs.
* Skips tests that are related to issue #1269 on windows.

Currently, I have dropped the Numpy 1.8.1 build as I need to compile my own conda packages like I was for wheels, but I will get this working asap. For now this should be merged so that Travis goes green.

### CODE REVIEW

**Comment by Cadair** [comment]
@coveralls YOU ARE BACK.

**Comment by Cadair** [comment]
WHY!

**Comment by ayshih** [comment]
Indeed, why =P

**Comment by Cadair** [comment]
I DON'T KNOW! :@

File: .travis.yml [hunk 1]
```diff
@@ -1,76 +1,68 @@
-language: python
+language: c
```
Comment (dpshelio) [inline .travis.yml]: Why is `c` instead of `python`?  Would be good to document this for the future?

File: .travis.yml [hunk 2]
```diff
@@ -1,76 +1,68 @@
...
+os:
+    - linux
+    - osx
 
-virtualenv:
-     system_site_packages: false
+cache:
+    directories:
+        $HOME/miniconda/pkgs
+        /Users/travis/miniconda3/pkgs
```
Comment (dpshelio) [inline .travis.yml]: is it `$HOME` == `/Users/travis`? What's the difference between `miniconda` and `miniconda3`

File: .travis.yml (hunk 1 above)
Comment (Cadair) [inline .travis.yml]: because we don't need to use Travis's python envs, and they don't work on OS/X anyway.

File: .travis.yml (hunk 2 above)
Comment (Cadair) [inline .travis.yml]: One is the Mac path and one is the linux one, it is the default install location for miniconda. These directives do not actually do anything at the moment.

File: .travis.yml [hunk 3]
```diff
@@ -1,76 +1,68 @@
...
-          env: TEST_MODE='astropy-dev'
-        - python: 2.7
-          env: TEST_MODE='sphinx'
-        - python: 2.7
-          env: TEST_MODE='online'
+    include:
+#         - os: linux
+#           env: NUMPY_VERSION=1.8.1
+         - os: linux
+           env: PANDAS_VERSION=0.14.1
```
Comment (dpshelio) [inline .travis.yml]: when `PANDAS_VERSION` or `LATEST_PANDAS` is used? I've just find an instance for the first, but not for the last.

Comment (Cadair) [inline .travis.yml]: LATEST_PANDAS is just the place where the latest version of pandas should be set. Same for numpy, it's just a naming thing to make it easier for people to modify.

File: continuous-integration/travis/install_linux.sh [hunk 4]
```diff
@@ -0,0 +1,23 @@
+###############################################################################
+# Install Linux Packages
+###############################################################################
+sudo apt-get upgrade
+sudo apt-get install -qq libatlas-dev liblapack-dev gfortran
+if [[ $TEST_MODE == 'sphinx' ]]; then sudo apt-get install graphviz texlive-latex-extra dvipng; fi
+
+# Install more upto date openjpeg library.
+wget http://openjpeg.googlecode.com/files/openjpeg-1.5.0-Linux-x86_64.tar.gz
+sudo tar -xvf openjpeg-1.5.0-Linux-x86_64.tar.gz --strip-components=1 -C /
```
Comment (dpshelio) [inline continuous-integration/travis/install_linux.sh]: Does this install `openjpeg`?

Comment (dpshelio) [inline continuous-integration/travis/install_linux.sh]: Ok, I see it, it just put it in the top directory... I thought it needed to compile or something...

File: .travis.yml (hunk 3 above)
Comment (dpshelio) [inline .travis.yml]: But is it used?

**Comment by dpshelio** [comment]
It looks all good for me :+1:
Thanks @Cadair !!!!

**Comment by Cadair** [comment]
@ayshih This now closes #1269

**Comment by ayshih** [comment]
What's the verdict going to be from Coveralls?

**Comment by Cadair** [comment]
Ok, coveralls is still being a pain, but we should merge this now so that the PRs for the meeting this evening can run properly.

**Comment by ayshih** [comment]
:+1: and merging, coveralls be damned

**Comment by Cadair** [comment]
And there was much rejoicing.

**Comment by Cadair** [comment]
As this is now in, I have turned on AppVeyor for all new builds as well as Travis.

**Pull request merged**